File: 451-SortCharactersByFrequency/451-SortCharactersByFrequency.py
# Last updated: 07/04/2026, 11:39:01
class Solution:
    def frequencySort(self, s: str) -> str:
        from collections import Counter
        
        count = Counter(s)
        
        k = sorted(count.items() ,key=lambda x:x[1],reverse=True)
        
        
        ans = ""
        for i in k :
            q ,w = i 
            ans +=(q*w)
            
            
        return ans 
            
        
        
        

        
        

File: 451-SortCharactersByFrequency/451-SortCharactersByFrequency.cpp
// Last updated: 07/04/2026, 11:38:55
class Solution {
public:
    string frequencySort(string s) {
        map<char,int> freq;


        for ( char c : s) freq[c]++;
        priority_queue<pair<int,char>> pq;
        for ( auto[c,i]: freq){
            pq.push({i,c});
        }
        string ans;
        
        while(!pq.empty()){
            auto [i,c] = pq.top();
            while(i--) ans+=c;
            pq.pop();
        }return ans;

    }
};